frontend/pages/admin/charities/[charityId]/edit.tsx

```diff
@@ -0,0 +1,91 @@
+import { Box, Container } from '@mui/system';
+import Head from 'next/head';
+import useSWR from 'swr';
+import CharityForm, { charitySchema } from '../../../../components/charities/form/CharityForm';
+import { containerSx } from '../../../../styles/admin/charities/createStyles';
+import { CharityData, CharityFormData, CharityPutData } from '../../../../types/charity';
+import * as Yup from 'yup';
+import api from '../../../../frontendApis';
+import { useRouter } from 'next/router';
+import { Stack, Typography } from '@mui/material';
+import { errorStackSx } from '../../../../styles/admin/charities/editStyles';
+import useAdminLoginCheck from '../../../../hooks/useAdminLogInCheck';
+
+const CharityEdit = () => {
+  useAdminLoginCheck();
+  const router = useRouter();
+
+  // Note that the router query parameters will all be undefined on initial render.
+  const { query } = router;
+  const charityId = query.charityId ? Number(query.charityId) : undefined;
+  const { data: initialValues, error } = useSWR<CharityFormData | undefined>([charityId], getCharityData);
+  const isLoading = !initialValues && !error;
+
+  const handleSubmit = (values: CharityFormData) => {
+    charitySchema
+      .validate(values)
+      .then((values: Yup.InferType<typeof charitySchema>) => {
+        const charityPostData: CharityPutData = {
+          ...values,
+        };
+
+        return api.charities.putCharity(charityId as number, charityPostData);
+      })
+      .then(() => {
+        router.push('/admin/charities');
+      });
+  };
+
+  return (
+    <Box>
+      <Head>
+        <title>Edit Charity</title>
+      </Head>
+
+      <Container sx={containerSx} component="main" maxWidth="sm">
+        {initialValues ? (
```

Comment:
Can we use guard clauses for this because the ternary conditions are quite complicated. 

Note to myself: Some of the stuff like loading and error page would also need to be done in future/ extracted out.